test/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Models/Propagated.test.cpp

```diff
@@ -0,0 +1,969 @@
+    using ostk::physics::coord::Velocity ;
+    using ostk::physics::Environment ;
+    using ostk::physics::env::obj::celest::Earth ;
+
+    using ostk::astro::trajectory::Orbit ;
+    using ostk::astro::trajectory::State ;
+
+    using ostk::astro::trajectory::orbit::models::Propagated ;
+
+    {
+
+        // Environment setup
+        const Environment environment = Environment::Default() ;
+
+        const Shared<const Frame> gcrfSPtr = Frame::GCRF() ;
+        
+        // Current state and instant setup
+
+        const Instant startInstant = Instant::DateTime(DateTime(2018, 1, 1, 0, 0, 0), Scale::UTC) ;
+
+        const State state = { startInstant, Position::Meters({ 0.0, 0.0, 0.0 }, gcrfSPtr), Velocity::MetersPerSecond({ 1.0, 0.0, 0.0 }, gcrfSPtr) };
+
+        // Orbital model setup
+        const Derived gravitationalParameter = Earth::Models::EGM2008::GravitationalParameter ;
+        const Length equatorialRadius = Earth::Models::EGM2008::EquatorialRadius ;
+
+        const Propagated propagatedModel = { state, startInstant, gravitationalParameter, equatorialRadius, Propagated::GravitationalPerturbationType::None, Propagated::AtmosphericPerturbationType::None, Propagated::ThirdBodyPerturbationType::None , false } ;
+
+        EXPECT_TRUE(typeid(propagatedModel.calculateStateAt(startInstant))==typeid(State)) ; 
+
+        EXPECT_TRUE(typeid(propagatedModel.calculateRevolutionNumberAt(startInstant))==typeid(Integer)) ; 
+
+    }
+
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+TEST (OpenSpaceToolkit_Astrodynamics_Trajectory_Orbit_Models_Propagated, StringFromPerturbationType)
+{
+
+    using ostk::core::types::Shared ;
+    using ostk::core::types::Real ;
+    using ostk::core::types::Integer ;
+    using ostk::core::types::String ;
+    using ostk::core::ctnr::Array ;
+    using ostk::core::ctnr::Table ;
+    using ostk::core::fs::Path ;
+    using ostk::core::fs::File ;
+
+    using ostk::math::obj::Vector3d ;
+
+    using ostk::physics::units::Length ;
+    using ostk::physics::units::Angle ;
+    using ostk::physics::units::Derived ;
+    using ostk::physics::time::Scale ;
+    using ostk::physics::time::Instant ;
+    using ostk::physics::time::Duration ;
+    using ostk::physics::time::Interval ;
+    using ostk::physics::time::DateTime ;
+    using ostk::physics::coord::Frame ;
+    using ostk::physics::coord::Position ;
+    using ostk::physics::coord::Velocity ;
+    using ostk::physics::Environment ;
+    using ostk::physics::env::obj::celest::Earth ;
+
+    using ostk::astro::trajectory::Orbit ;
+    using ostk::astro::trajectory::State ;
+
+    using ostk::astro::trajectory::orbit::models::Propagated ;
+
+    {
+        
+        EXPECT_TRUE(typeid(Propagated::StringFromGravitationalPerturbationType(Propagated::GravitationalPerturbationType::None))==typeid(String)) ; 
```

Comment:
```suggestion
        EXPECT_TRUE(typeid(Propagated::StringFromGravitationalPerturbationType(Propagated::GravitationalPerturbationType::None)) == typeid(String)) ; 
```
To be applied below as well. But in general, no need to use `typeid` as discussed above IMO